### ISSUE TITLE
Tempo and Pitch adjustment options with a button like design instead of falt text based

### ISSUE THREAD

**Issue by desmonk** [TeamNewPipe/NewPipe#8962]
### Checklist

- [X] I made sure that there are *no existing issues* - [open](https://github.com/TeamNewPipe/NewPipe/issues) or [closed](https://github.com/TeamNewPipe/NewPipe/issues?q=is%3Aissue+is%3Aclosed) - which I could contribute my information to.
- [X] I have read the [FAQ](https://newpipe.net/FAQ/) and my problem isn't listed.
- [X] I'm aware that this is a request for NewPipe itself and that requests for adding a new service need to be made at [NewPipeExtractor](https://github.com/TeamNewPipe/NewPipeExtractor/issues).
- [X] I have taken the time to fill in all the required details. I understand that the feature request will be dismissed otherwise.
- [X] This issue contains only one feature request.
- [X] I have read and understood the [contribution guidelines](https://github.com/TeamNewPipe/NewPipe/blob/dev/.github/CONTRIBUTING.md).

### Feature description

The interface here have option to adjust tempo and pitch but they have a flat text based layout instead of some kind of clickable buttons. With little bit of design change there, we can know visualize much better as for now they just look like video statistics menu.

### Why do you want this feature?

To enhance the visibility of the tempo and pitch functions giving users the ability to differentiate with ease by having more understanding about the app usability.

### Additional information

Just a simple layout change can do wonders for users. Below is just an example.
![IMG_21739813](https://user-images.githubusercontent.com/55331939/189597015-e5d4768d-b6a3-44cb-95c4-736fde8c3a80.PNG)

**Comment by opusforlife2** [comment]
Accessibility-enabled users would probably want easily distinguishable buttons too.

**Comment by u7280973** [comment]
Hey, could I work on this issue? :))

**Comment by opusforlife2** [comment]
@u7280973 Sure! Could you post a screenshot of your proposed design?

**Comment by u7280973** [comment]
Perhaps something like this?

<img width="350" alt="image" src="https://user-images.githubusercontent.com/110881895/195546883-1de53cd7-f2b8-4d46-86d7-1257483ebe9b.png">

**Comment by opusforlife2** [comment]
Looks nice, @u7280973! What about dark theme? And black theme as well, now that I think about it?

**Comment by u7280973** [comment]
Something like this? Black theme has the same design as dark theme.

<img width="396" alt="image" src="https://user-images.githubusercontent.com/110881895/195621789-946e60fb-04bc-4a9b-ac5a-cc8f5583d970.png">

**Comment by opusforlife2** [comment]
Awesome! Can the red colours be made the same for the seekbar and the buttons, though? The difference isn't noticeable in light theme, but looks weird in the dark theme.

Also, the dotted border is rounded in the dark theme. Should be the same across themes, no?

**Comment by u7280973** [comment]
Sounds good, I'll fix the dark theme one. And yeah, I think the rounder border looks a bit better, would you agree? Here's what I'm thinking for light and dark theme: 
<img width="340" alt="image" src="https://user-images.githubusercontent.com/110881895/195623263-e8d3296c-f5ba-4fb8-ba50-09d4c2cf99ed.png">
<img width="344" alt="image" src="https://user-images.githubusercontent.com/110881895/195623342-b237f6fc-c9eb-4e78-86f1-a4033c810fba.png">

**Comment by opusforlife2** [comment]
This UI looks like you're supposed to choose between Step and Speed.

**Comment by opusforlife2** [comment]
What I meant is: the UI indicates that you're supposed to switch between 2 menus, 'Speed' and 'Step'. When you say that Speed itself will be a 3-state toggle, then it becomes highly non-intuitive for the user. Choose one. Multi-state toggles, or switchable menus.

**Comment by Stypox** [comment]
You have a point there. Most of the times one wants to just change one of the sliders at a time.

I don't think that exact layout is the best one, but we can work on it. What do you think of this one?

<img src="https://user-images.githubusercontent.com/36421898/198321630-00d378d8-3867-43d9-a16a-b071615c5660.jpg" height="200px"/>

Tapping on the "±5%" would bring up a popup menu allowing to choose the step percent (assuming this is an action that the user performs really rarely).
Tapping on the "Tempo" would bring up a popup menu allowing to switch between different types of sliders (tempo, pitch - tones, pitch - percent). I don't know if this is easily discoverable.

**Comment by opusforlife2** [comment]
> Tapping on the "Tempo" would bring up a popup menu 

This is also what I had in mind. I don't think this will be difficult to discover. Android has plenty of popup menus indicated by a down arrow. In fact, to make it easier, the entire row/area should be made tappable, so that the user is more likely to accidentally tap it and discover the menu.

Having said that, I would like its position more towards the top-right, where Sameen's pic has Step and Speed.

Also, the +-5% button is hard to discover. It could have a border or some other colour to differentiate it.

>  "unhook" should be given a more explanatory name

"Unlink speed and pitch"

**Comment by Stypox** [comment]
I agree with all of the things you said. Maybe to make the ±5% button more discoverable we could just add a down arrow there, too.

**Comment by opusforlife2** [comment]
Let's try that and compare screenshots.

**Comment by Stypox** [comment]
<img src="https://user-images.githubusercontent.com/36421898/198340181-789eb706-1df3-4879-82b1-15c793bb5a65.jpg" height="300px"/>

**Comment by opusforlife2** [comment]
Looks great!

**Comment by Stypox** [comment]
What does step mean? Why would step be a slider?

**Comment by opusforlife2** [comment]
I don't think that would help. Step is for modifying the values shown in the sliders, not actually affecting video playback. So at least one slider should stay visible while the step size is changed for the effect to be clear.

**Comment by opusforlife2** [comment]
> Isn't that problem still persistent in case of https://github.com/TeamNewPipe/NewPipe/issues/8962#issuecomment-1293747666?

No, there you can easily see the step size change being applied as you tap the arrows. You don't need to change menus if you don't like the current value, you merely tap the % button and select a new one.

---

One thing I'm unaware about is if these are standard Android behaviours. The closer we are to common and standardised behaviour, the better.

**Comment by opusforlife2** [comment]
I like this one. ^

**Comment by tsilvs** [comment]
It might be more convenient to have fixed playback tempo rate options as buttons and/or as equal value steps.

For example, here's a playback speed controls menu from "Global Speed" browser extension:

![Global Speed extension menu](https://github.com/TeamNewPipe/NewPipe/assets/26058495/4ec041c8-8a8d-47a7-be11-45f2c4c82e6f)

These buttons are customizable presets:

![4 rows](https://github.com/TeamNewPipe/NewPipe/assets/26058495/506a842c-ee18-42e4-9b29-cb2a18a4a520)

![1 row](https://github.com/TeamNewPipe/NewPipe/assets/26058495/78285fec-99eb-47d4-8d4f-8e81d7466350)

![Changed menu](https://github.com/TeamNewPipe/NewPipe/assets/26058495/a9e047ce-79ca-475f-ba86-c7ec88ccf8bf)

Similar example with presets approach is AntennaPod podcast listening app, which has these "tempo / playback speed" controls:

![AntennaPod podcast listening app "tempo / playback speed" controls](https://github.com/TeamNewPipe/NewPipe/assets/26058495/5399fd44-f4b2-4914-ac4f-bf5003d56978)
